istioctl/cmd/istioctl/main.go

```diff
@@ -171,6 +178,9 @@ See https://istio.io/docs/reference/ for an overview of Istio routing.
 					return err
 				}
 				var rev string
+				if deprecated, _ := deprecatedTypes[config.Type]; deprecated {
+					c.Printf("Warning: %s is deprecated and may not be supported in future Istio versions (%s).\n", config.Type, config.Name)
```

Comment:
s/may/will 

Also, any chance we cab give a hard deprecation date/version here? Is 1.0 too soon?